tsdb/head_bench_test.go

```diff
@@ -79,6 +86,85 @@ func BenchmarkHeadStripeSeriesCreate_PreCreationFailure(b *testing.B) {
 	}
 }
 
+func BenchmarkHead_WalCommit(b *testing.B) {
+	seriesCounts := []int{100, 1000, 10000}
+	series := genSeries(10000, 10, 0, 0) // Only using the generated labels.
+
+	appendSamples := func(b *testing.B, app storage.Appender, seriesCount int, ts int64) {
+		var err error
+		for i, s := range series[:seriesCount] {
+			var ref storage.SeriesRef
+			if i%2 == 0 {
+				ref, err = app.Append(ref, s.Labels(), ts, float64(ts))
+			} else {
+				h := &histogram.Histogram{
+					Count:         7 + uint64(ts*5),
+					ZeroCount:     2 + uint64(ts),
+					ZeroThreshold: 0.001,
+					Sum:           18.4 * rand.Float64(),
+					Schema:        1,
+					PositiveSpans: []histogram.Span{
+						{Offset: 0, Length: 2},
+						{Offset: 1, Length: 2},
+					},
+					PositiveBuckets: []int64{ts + 1, 1, -1, 0},
+				}
+				ref, err = app.AppendHistogram(ref, s.Labels(), ts, h, nil)
+			}
+			require.NoError(b, err)
+
+			_, err = app.AppendExemplar(ref, s.Labels(), exemplar.Exemplar{
+				Labels: labels.FromStrings("trace_id", strconv.Itoa(rand.Int())),
+				Value:  rand.Float64(),
+				Ts:     ts,
+			})
+			require.NoError(b, err)
+		}
+	}
+
+	for _, seriesCount := range seriesCounts {
+		b.Run(fmt.Sprintf("%d series", seriesCount), func(b *testing.B) {
+			for _, commits := range []int64{1, 2} {
```

Comment:
Similarly here, I don't understand why we're doing 1 or 2 commits 🤔

Reply:
If I'm not wrong this is acting as a warm up, can you help @krajorama?

Reply:
I suggested that to measure the case where series are created in the head vs the case where the series already exists.